hashintel/hash
# Add theming proof-of-concept

## 🌟 What is the purpose of this PR?

This PR introduces `BLOCK_BASED_ENTITY_EDITOR` feature flag, which helps us develop a new entity editor without breaking the existing one. This first iteration focuses on passing styling down into the new `<EntityPropertiesBlock />` component.




## ⚠️ Known issues

- ~~There are quite a lot of warnings in the Terminal console.~~ Fixed in https://github.com/hashintel/hash/pull/441.



## 🐾 Next steps

- Continue shaping the theme object (improve types)
- Move patches to `blockprotocol` package
- Sanitize the theme object (remove non-serializable parts of the MUI Theme to better simulate sandbox boundary)



## 🔍 What does this change?




- New entity editor behind the `BLOCK_BASED_ENTITY_EDITOR` feature flag
- Changes in `blockprotocol` patch (added `styling.d.ts`)
- File structure improvements in `packages/hash/frontend/src/pages/[accountId]/entities` (using `kebab-case` and local imports)
- New `LocalReactBlockContainer` component (enables local React-based blocks with a separate tree)
- `entityName` was renamed to `guessEntityName`, JSDoc was added for clarity

### 📜 Does this require a change to the docs?



- Yes, on `blockprotocol.org`. We can do this when updating the `blockprotocol` package.

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1200637993232800/1201487917465427/f) _(internal)_

## 🛡 What tests cover this?

- local manual tests

## ❓ How to test this?



1.  Checkout the branch / view the deployment
1.  Run `BLOCK_BASED_ENTITY_EDITOR=true yarn dev`
1.  Open a page you have access to
1.  Change URL from `http://localhost:3000/x/y` to `http://localhost:3000/x/entities/y`.
1.  Observe a new page with a styled MUI `<Button />` and `<Typography />`
1.  Open `packages/hash/frontend/src/pages/[accountId]/entities/[entity-id].page/block-based-entity-editor.tsx` and remove `theme={theme}`
1.  Confirm that the theme from the embedding app no longer applies

## 📹 Demo

<img width="943" alt="Screenshot 2022-03-29 at 23 04 51" src="https://user-images.githubusercontent.com/608862/160714995-3393d085-81e5-4bb7-8fbf-ecfddb12a1d9.png">
